experiments/filtervisualizer.py

```diff
@@ -0,0 +1,112 @@
+import matplotlib as plt
+
+from fastai.conv_learner import *
+
+
+class SaveFeatures():
+    def __init__(self, module):
+        self.hook = module.register_forward_hook(self.hook_fn)
+
+    def hook_fn(self, module, _input, output):
+        self.features = output
+
+    def close(self):
+        self.hook.remove()
+
+
+class FilterVisualizer():
+    def __init__(self,
+                 model,
+                 size=56,
+                 upscaling_steps=12,
+                 upscaling_factor=1.2,
+                 cpu=False):
+        self.size, self.upscaling_steps, self.upscaling_factor = size, upscaling_steps, upscaling_factor
+        self.model = model
+        set_trainable(self.model, False)
+        self.cpu = cpu
+
+    def visualize(self,
+                  layer,
+                  conv_filter,
+                  lr=0.1,
+                  opt_steps=20,
+                  print_losses=False,
+                  layer_name_plot=None,
+                  blur=None, ):
+        sz = self.size
+        img = (np.random.random((sz, sz, 3)) * 20 + 128.) / 255.
+        activations = SaveFeatures(layer)  # register hook
+
+        for i in range(self.upscaling_steps
+                       ):  # scale the image up upscaling_steps times
+
+            if i > self.upscaling_steps / 2:
+                opt_steps_ = int(opt_steps * 1.3)
+            else:
+                opt_steps_ = opt_steps
+            train_tfms, val_tfms = tfms_from_model(self.model, sz)
```

Comment:
This is done in the `get_transformed` and `most_activated` methods as well. How about making these instance attributes?

Reply:
Oh you are right. I overlooked that I'm afraid. I will change that!

Reply:
Thanks!